content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md

```diff
@@ -0,0 +1,207 @@
+
+    @StreamListener(Processor.INPUT)
+    @SendTo(Processor.OUTPUT)
+    public UsageCostDetail processUsageCost(UsageDetail usageDetail) {
+    UsageCostDetail usageCostDetail = new UsageCostDetail();
+    usageCostDetail.setUserId(usageDetail.getUserId());
+    usageCostDetail.setCallCost(usageDetail.getDuration() * this.ratePerSecond);
+    usageCostDetail.setDataCost(usageDetail.getData() * this.ratePerMB);
+    return usageCostDetail;
+    }
+    }
+    ```
+
+#### Testing
+
+You need to set
+
+```
+spring.cloud.stream.bindings.input.destination=test-usage-detail
+spring.cloud.stream.bindings.output.destination=test-usage-cost
+server.port=9091
+```
+
+You can run the standalone `UsageCostProcessor` processor application as,
+
+```
+java -jar usage-cost-processor-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-detail --spring.cloud.stream.bindings.output.destination=test-usage-cost --server.port=9091
+```
+
+### Sample Sink
+
+#### Initialzr
+
+1. Visit the [Spring Initialzr site](https://start.spring.io/).
+1. Select the latest release of spring boot.
+1. Create a new Maven project with a Group name of `io.spring.dataflow` and an Artifact name of `usage-cost-logger`.
+1. In the Dependencies text box, type `Rabbitmq` to select the RabbitMQ binder dependency.
+1. In the Dependencies text box, type `Cloud Stream` to select the Spring Cloud Stream dependency.
+1. Click the Generate Project button.
+1. Unzip the usage-cost-logger.zip file and import the project into your favorite IDE.
+
+##### Initialzr Shortcut
+
+Another option instead of using the UI to initialize your project you can do the following:
+
+1. Click the [here](https://start.spring.io/starter.zip?fakeusernameremembered=&fakepasswordremembered=&type=maven-project&language=java&bootVersion=2.1.4.RELEASE&baseDir=usage-cost-logger&groupId=io.spring.dataflow.sample&artifactId=usage-cost-logger&name=usage-cost-logger&description=Sample+project+for+Spring+Cloud+Stream+Sink&packageName=io.spring.dataflow.sample&packaging=jar&javaVersion=1.8&inputSearch=&style=amqp&style=cloud-stream) to download the preconfigured usage-cost-logger.zip.
+2. Unzip the usage-cost-logger.zip file and import the project into your favorite IDE
+
+#### Biz Logic
+
+1.  In your favorite IDE create the `io.spring.dataflow.sample.domain` package.
+1.  Create a `UsageCostDetail` class in the `io.spring.dataflow.sample.domain` using using your favorite IDE that looks like the contents in [UsageCostDetail.java](https://github.com/spring-cloud/spring-cloud-dataflow-samples/blob/master/dataflow-website/stream-developer-guides/streams/usage-cost-processor/src/main/java/com/example/demo/domain/UsageCostDetail.java).
+    This `UsageCostDetail` model contains `userId`, `callCost` and `dataCost`.
+1.  Create the `Sink` application that receives the `UsageCostDetail` from the previously created `processor` and logs it.
+
+```java
+@EnableBinding(Sink.class)
+public class UsageCostLogger {
```

Comment:
Its unclear that we want to create a class. May want to reword this to something like:
"In the io.spring.dataflow package create a UsageCostLogger class using your favorite IDE that looks like the contents below." It's unclear where to create the class as well.